plugins/terraform/data_source_layer0_deploy.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/cli/client"
+)
+
+func dataSourceLayer0Deploy() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceLayer0DeployRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func datasourceLayer0DeployRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(client.Client)
+
+	deployName := d.Get("name").(string)
+	version := d.Get("version").(string)
+
+	deployID, err := resolveTags(client, deployName, map[string]string{
```

Comment:
[nit] this code block does't read well - maybe do something like:
```
params := map[string]string{
    "type": "deploy",
    "version": version,
}

deployID, err := resolveTags(client, deployName, params)
if err != nil {
    return err
}
...
```

(this applies to all entities)